File: 1759.py
from itertools import groupby

class Solution:
    def countHomogenous(self, s: str) -> int:
        m = int(1e9) + 7
        res = 0
        for g in groupby(s):
            n = len(tuple(g[1]))
            res += n * (n + 1) // 2
        return res % m

File: sol/1759.cpp
struct Solution {
    int countHomogenous(const string &s) {
        int m = 1'000'000'007, res = 0, ptrl = 0, ptrr;
        long long n, comb;
        while (ptrl < s.size()){
            ptrr = ptrl + 1;
            while (ptrr < s.size() && s[ptrl] == s[ptrr]){
                ++ptrr;
            }
            n = ptrr - ptrl;
            if (n & 1){
                comb = (((n + 1) >> 1) * n) % m;
            } else {
                comb = ((n >> 1) * (n + 1)) % m;
            }
            res = (res + comb) % m;
            ptrl = ptrr;
        }
        return res;
    }
};
